src/resources/class-modifiers-for-apis.md

```diff
@@ -0,0 +1,359 @@
+so when you declare a class in Java, C#, or other languages,
+you effectively have the same limitation.
+
+## The `final` modifier
+
+If you want all of the restrictions of both `interface` and `base`,
+you can mark a class or mixin class `final`.
+This prevents anyone outside of your library from creating
+any kind of subtype of it:
+no using it in `implements`, `extends`, `with`, or `on` clauses.
+
+This is the most restrictive for users of the class.
+All they can do is construct it (unless it's marked `abstract`).
+In return, you have the fewest restrictions as the class maintainer.
+You can add new methods, turn constructors into factory constructors, etc.
+without worrying about breaking any downstream users.
+
+## The `sealed` modifer
+
+The last modifier, `sealed`, is special.
+It exists primarily to enable [exhaustiveness checking][] in pattern matching.
+If a switch has cases for every direct subtype of a type marked `sealed`,
+then the compiler knows the switch is exhaustive.
+
+[exhaustiveness checking]: /language/branches#exhaustiveness-checking
+
+```dart
+// amigos.dart
+abstract class Amigo {}
+class Lucky extends Amigo {}
+class Dusty extends Amigo {}
+class Ned extends Amigo {}
+
+String lastName(Amigo amigo) =>
+    switch (amigo) {
+      case Lucky _ => 'Day';
+      case Dusty _ => 'Bottoms';
+      case Ned _   => 'Nederlander';
+    }
+```
+
+This switch has a case for each of the subtypes of `Amigo`.
+The compiler knows that every instance of `Amigo` must be an instance of one
+of those subtypes, so it knows the switch is safely exhaustive and doesn't
+require any final default case.
+
+For this to be sound, the compiler enforces two restrictions:
+
+1.  The sealed class can't itself be directly constructible.
+    Otherwise, you could have an instance of `Amigo` that isn't
+    an instance of *any* of the subtypes.
+    So every `sealed` class is implicitly `abstract` too.
+
+2.  Every direct subtype of the sealed type must be in the same library
+    where the sealed type is declared.
+    This way the compiler can find them all and know that there aren't
+    other hidden subtypes floating around that would not match any of the cases.
+
+The second restriction is similar to `final`.
+Like `final`, it means that a class marked `sealed` can't be directly
+extended, implemented, or mixed in outside of the library where it's declared.
+But, unlike `base` and `final`, there is no *transitive* restriction:
+
+```dart
+// amigo.dart
+abstract class Amigo {}
+class Lucky extends Amigo {}
+class Dusty extends Amigo {}
+class Ned extends Amigo {}
+
+// other.dart
+
+// This is an error:
+class Bad extends Amigo {}
+
+// But these are both fine:
+class OtherLucky extends Lucky {}
+class OtherDusty implements Dusty {}
+```
+
+Of course, if you *want* the subtypes of your sealed type
+to be restricted as well, you can get that by marking them
+using `interface`, `base`, `final`, or `sealed`.
+
```

Comment:
I would make this it's own section (it'd be nice to point people to directly). Either top level or a subsection of sealed:

```suggestion

## `sealed` versus `final`

```